src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -1194,6 +1194,7 @@ trait Types
       else super.safeToString
     override def narrow: Type = this
     override def kind = "ThisType"
+    override def boundSyms = if (sym.isInstanceOf[StubSymbol]) emptySymbolSet else super.boundSyms
```

Comment:
I've removed the override for now. I think we could actually generalize it to:

```scala
class SubType {
  override def boundSyms = emptySymbolSet
}
```

Along with overrides to pick that:

```
abstract case class ThisType {
  override def boundSyms = super[SubType].boundSyms
}
...
```

But let's rely on `underlying.boundSyms` for now.